### PR TITLE
Migrate to AssemblyLine phase 1

### PR DESCRIPTION
**Pull request by nonprofittechy**
This is part 1 of migration to the AssemblyLine package. Phase 2 will include migration to the ALDocument class

This includes:

1. Changing dependencies from MAVC=>AL. Changing dependency on AssemblyLineToolbox=>ALToolbox. **Note: this makes the 209A package installable on a new server! Important.**
2. Added some functions to ALToolbox that were in the old virtual_court_support.py in MAVC. Therefore, this depends on ALToolbox > 0.9 being installed.
3. Renaming variables, such as courts[0] and basic_questions_intro_screen to the AL equivalent.
4. Renamed objects from VCIndividual to ALIndividual
5. Added gender_fields to ALIndividual class in AssemblyLine repo. Therefore, this depends on AssemblyLine >= 2.0.15 being installed
6. Made use of name_fields and gender_fields method throughout, to reduce strings that need to be translated.
7. Migrated several list collects to one-at-a-time questions for compatibility with name_fields() method.
8. Added escape.js and escape.css directly into this package. It would be reasonable to move this to ALToolbox instead in the future.

I ran through two branches to try to use every new screen.

I did uncover at least one bug in existing 209A, #61 and confirmed it is not related to this code. Overall this is should not change the behavior of the  209A. This should be merged quickly to make the 209A installable on a new server.

### CODE REVIEW

**Comment by plocket** [comment]
Tests so far: I believe I tried this interview at c3aab93. Wasn't aware there were new changes. I included minor and major children and a previous case with them, as well as a previous case with the defendant themselves. I got through to the end, though I did not test out review screens.

When I attempted to hit 'Undo' to get back to the beginning (for reasons), I hit an error when I hit 'undo' on the page ID `what to expect`, I got this error:
```
LazyNameError: name 'other_parties[0].gender_female_label' is not defined
Log
Traceback (most recent call last):
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/webapp/server.py", line 11106, in index
    content = as_html(interview_status, url_for, debug_mode, url_for('index', **index_params), validation_rules, the_field_errors, the_progress_bar, steps - user_dict['_internal']['steps_offset'])
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/base/standardformatter.py", line 1262, in as_html
    fieldlist.append('                <div ' + style_def + data_def + 'class="form-group row' + side_note_parent + req_tag + field_class + class_def + '"><label' + label_for + ' class="col-md-4 col-form-label da-form-label datext-right">' + helptext_start + markdown_to_html(status.labels[field.number], trim=True, status=status, strip_newlines=True) + helptext_end + '</label><div class="col-md-8 dafieldpart">' + input_for(status, field) + '</div>' + side_note + '</div>\n')
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/base/standardformatter.py", line 2313, in input_for
    other_options += '>' + markdown_to_html(str(pair['label']), status=status, escape='option', trim=True, do_terms=False) + '</option>'
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/base/core.py", line 4477, in __str__
    return(str(self.content))
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/base/core.py", line 4438, in content
    raise LazyNameError("name '" + str(self.instanceName) + "' is not defined")
docassemble.base.error.LazyNameError: name 'other_parties[0].gender_female_label' is not defined
```

**Comment by BryceStevenWilley** [comment]
> When I attempted to hit 'Undo' to get back to the beginning (for reasons), I hit an error when I hit 'undo' on the page ID what to expect, I got this error: ...

I ran into this a while back and Jonathan helped my out on Slack: you just need to wrap the `template` object in a `str` wherever it is.  https://docassemble.slack.com/archives/C7791ATUJ/p1617064981363700?thread_ts=1617031665.354200&cid=C7791ATUJ

**Comment by nonprofittechy** [comment]
Thanks--I patched this in suffolklitlab/AssemblyLine and installed the new version on apps-dev

File: docassemble/MA209AProtectiveOrder/data/questions/209A_page_1.yml [hunk 1]
```diff
@@ -1108,7 +1100,9 @@ buttons:
   - Restart: restart
 attachment code: complaint_209A_labeled_page10003_attachment
 ---
-need: complaint_209A_labeled_page10003
+need: 
+  - complaint_209A_labeled_page10003
+  - defendant_weapons_safe_for_pdf
```
Comment (plocket) [inline docassemble/MA209AProtectiveOrder/data/questions/209A_page_1.yml:1105]: Would this document need to be recalculated if these values changed?

**Review by plocket — APPROVED** [review]
Other than the question about that one `needs` specifier, this looks good to me. Whatever decision is made about that will be fine. If more is changed, I can take another look.

**Pull request merged**